### PR TITLE
Add `OR REPLACE`, `IF NOT EXISTS` and `IF EXISTS` to create and drop commands for indexes and constraints

### PR DESCRIPTION
**Pull request by Hunterness**
Companion to https://github.com/neo-technology/neo4j/pull/6677, which needs to go in first

### CODE REVIEW

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 1]
```diff
@@ -35,6 +35,12 @@ Replacement syntax for deprecated and removed features are also indicated.
 | `SHOW ROLE name PRIVILEGES` | Functionality | Updated | Can now handle multiple roles, `SHOW ROLES n1, n2, ... PRIVILEGES`
 | `SHOW USER name PRIVILEGES` | Functionality | Updated | Can now handle multiple users, `SHOW USERS n1, n2, ... PRIVILEGES`
 | `DEFAULT GRAPH` | Syntax | Added | New optional part of the Cypher commands for <<administration-security-administration-database-privileges, database privileges>>
+| `CREATE OR REPLACE INDEX name FOR ...` | Syntax | Added | Can now drop and recreate an index in one command, based on the name of the index
+| `CREATE INDEX [name] IF NOT EXISTS FOR ...` | Syntax | Added | Makes index creation idempotent, if an index with the name or schema already exist no error will be thrown
```
Comment (recrwplay) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
| `CREATE INDEX [name] IF NOT EXISTS FOR ...` | Syntax | Added | Makes index creation idempotent. If an index with the name or schema already exists no error will be thrown
```

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 2]
```diff
@@ -35,6 +35,12 @@ Replacement syntax for deprecated and removed features are also indicated.
...
+| `DROP INDEX name IF EXISTS` | Syntax | Added | Makes index deletion idempotent, if no index with the name exists no error will be thrown
```
Comment (recrwplay) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
| `DROP INDEX name IF EXISTS` | Syntax | Added | Makes index deletion idempotent. If no index with the name exists no error will be thrown
```

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 3]
```diff
@@ -35,6 +35,12 @@ Replacement syntax for deprecated and removed features are also indicated.
...
+| `CREATE OR REPLACE CONSTRAINT name ON ...` | Syntax | Added | Can now drop and recreate a constraint in one command, based on the name of the constraint
+| `CREATE CONSTRAINT [name] IF NOT EXISTS ON ...` | Syntax | Added | Makes constraint creation idempotent, if a constraint with the name or type and schema already exist no error will be thrown
```
Comment (recrwplay) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
| `CREATE CONSTRAINT [name] IF NOT EXISTS ON ...` | Syntax | Added | Makes constraint creation idempotent. If a constraint with the name or type and schema already exists no error will be thrown
```

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 4]
```diff
@@ -35,6 +35,12 @@ Replacement syntax for deprecated and removed features are also indicated.
...
+| `DROP CONSTRAINT name IF EXISTS` | Syntax | Added | Makes constraint deletion idempotent, if no constraint with the name exists no error will be thrown
```
Comment (recrwplay) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
| `DROP CONSTRAINT name IF EXISTS` | Syntax | Added | Makes constraint deletion idempotent. If no constraint with the name exists no error will be thrown
```

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc [hunk 5]
```diff
@@ -24,16 +24,22 @@ This section comprises a glossary of all the keywords -- grouped by category and
...
+|<<administration-constraints-syntax, CREATE OR REPLACE CONSTRAINT existence ON (n:Label) ASSERT exists(n.property)>>  | Schema   | Create or replace a constraint ensuring that all nodes with a particular label have a certain property.
+|<<administration-constraints-syntax, CREATE CONSTRAINT [node_key] [IF NOT EXISTS] ON (n:Label) ASSERT (n.prop1, ..., n.propN) IS NODE KEY>>  |  Schema | Create a constraint ensuring all nodes with a particular label have all the specified properties and that the combination of property values is unique; i.e. ensures existence and uniqueness.
+|<<administration-constraints-syntax, CREATE OR REPLACE CONSTRAINT node_key ON (n:Label) ASSERT (n.prop1, ..., n.propN) IS NODE KEY>>  |  Schema | Create or replace a constraint ensuring all nodes with a particular label have all the specified properties and that the combination of property values is unique; i.e. ensures existence and uniqueness.
+|<<administration-constraints-syntax, CREATE CONSTRAINT [existence] [IF NOT EXISTS] ON ()-[r:REL_TYPE]-() ASSERT exists(r.property)>>  | Schema   | Create a constraint ensuring that all relationship with a particular type have a certain property.
```
Comment (recrwplay) [inline cypher/cypher-docs/src/docs/dev/glossary.asciidoc]:
```suggestion
|<<administration-constraints-syntax, CREATE CONSTRAINT [existence] [IF NOT EXISTS] ON ()-[r:REL_TYPE]-() ASSERT exists(r.property)>>  | Schema   | Create a constraint ensuring that all relationships with a particular type have a certain property.
```

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc [hunk 6]
```diff
@@ -24,16 +24,22 @@ This section comprises a glossary of all the keywords -- grouped by category and
...
+|<<administration-constraints-syntax, CREATE OR REPLACE CONSTRAINT existence ON ()-[r:REL_TYPE]-() ASSERT exists(r.property)>>  | Schema   | Create or replace a constraint ensuring that all relationship with a particular type have a certain property.
```
Comment (recrwplay) [inline cypher/cypher-docs/src/docs/dev/glossary.asciidoc]:
```suggestion
|<<administration-constraints-syntax, CREATE OR REPLACE CONSTRAINT existence ON ()-[r:REL_TYPE]-() ASSERT exists(r.property)>>  | Schema   | Create or replace a constraint ensuring that all relationships with a particular type have a certain property.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala [hunk 7]
```diff
@@ -65,6 +65,13 @@ class SchemaIndexTest extends DocumentingTestBase with QueryStatisticsTestSuppor
       optionalResultExplanation = "Note that the index name needs to be unique. ",
       assertions = _ => assertIndexWithNameExists("index_name", "Person", List("surname"))
     )
+    testQuery(
+      title = "Create a single-property index only if it does not already exist",
+      text = "If it is unknown if an index exists or not but we want to make sure it does, we add the `IF NOT EXISTS`.",
```
Comment (recrwplay) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala]:
```suggestion
      text = "If it is unknown if an index exists or not but we want to make sure it does, we add `IF NOT EXISTS`.",
```

**Review by recrwplay — COMMENTED** [review]
Everything looks like it is appearing where it should. Just a few very minor items.

**Comment by Hunterness** [comment]
@recrwplay @SaschaPeukert Do you want to re-review this pr before its merged?

**Review by recrwplay — APPROVED** [review]
All looks good to me!

**Review by SaschaPeukert — APPROVED** [review]

**Pull request merged**